includes/html/api_functions.inc.php

```diff
@@ -1944,14 +1944,17 @@ function add_device_group(Illuminate\Http\Request $request)
         return api_error(422, $v->messages());
     }
 
+    $deviceGroup = DeviceGroup::make(['name' => $data['name'], 'type' => $data['type'], 'desc' => $data['desc']]);
+
     // Only use the rules if they are able to be parsed by the QueryBuilder
-    $query = QueryBuilderParser::fromJson($data['rules'])->toSql();
-    if (empty($query)) {
-        return api_error(500, "We couldn't parse your rule");
+    if ($data['type'] == 'dynamic') {
```

Comment:
Hmm, actually static groups can have rules, they are just unused until you change the type to dynamic.  We should check `if (! empty($data['rules'])) {` instead of the type

Reply:
@murrant - Absolutely, if the desired behaviour is to allow the definition of rules for static groups then wrapping the QueryBuilderParser::fromJson() call and accompanying api_error validation in the conditional you've suggested would be the way to go. I'll drop this pull request, test it and submit a new one with the change included.

Reply:
You don't need to close this, just update it :)